src/main/flight/pid.c

```diff
@@ -510,13 +511,33 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
         if (axis != FD_YAW) {
             // apply filters
             float gyroRateFiltered = dtermNotchFilterApplyFn(dtermFilterNotch[axis], gyroRate);
-            gyroRateFiltered = dtermLpfApplyFn(dtermFilterLpf[axis], gyroRateFiltered);
+            if (pidProfile->dterm_filter_style == KD_FILTER_CLASSIC)
+            {
+                gyroRateFiltered = dtermLpfApplyFn(dtermFilterLpf[axis], gyroRateFiltered);
+            }
 
             const float rD = dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) * currentPidSetpoint - gyroRateFiltered;    // cr - y
-            // Divide rate change by deltaT to get differential (ie dr/dt)
-            float delta = (rD - previousRateError[axis]) / deltaT;
-
-            previousRateError[axis] = rD;
+            const float pureRD = dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) * getSetpointRate(axis) - gyroRateFiltered;    // cr - y
```

Comment:
You are calculating pureRD, event for CLASSIC and SP, what about to calculate here just the common base for both dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) and the rest add in appropridate cases?